platform/ios/src/MGLMapView.mm

```diff
@@ -3761,7 +3761,12 @@ - (void)selectAnnotation:(id <MGLAnnotation>)annotation animated:(BOOL)animated
         UIView <MGLCalloutView> *calloutView;
         if ([self.delegate respondsToSelector:@selector(mapView:calloutViewForAnnotation:)])
         {
-            calloutView = [self.delegate mapView:self calloutViewForAnnotation:annotation];
+            id providedCalloutView = [self.delegate mapView:self calloutViewForAnnotation:annotation];
+            if (providedCalloutView) {
+                NSAssert([providedCalloutView isKindOfClass:[UIView class]], @"callout view should be a kind of UIView");
```

Comment:
This should be an NSException, not an NSAssert. Now that the return value of `-mapView:calloutViewForAnnotation:` is no longer required to be a kind of UIView, returning something other than a UIView is a developer error, not an SDK error.